rc/base/ctags.kak

```diff
@@ -6,44 +6,53 @@
+        fi
+        cat "$namecache"
+    done
+} ctags-search -params ..1 %{ evaluate-commands %sh{
+    realpath() { ( path=$(readlink "$1"); cd "$(dirname "$1")"; printf "%s/%s\n" "$(pwd -P)" "$(basename "$1")" ) }
+    export tagname=${1:-${kak_selection}}
+    menu="${TMPDIR:-/tmp}/ctags-menu"
+    open='{'; close='}'
+    eval "set -- $kak_opt_ctagsfiles"
+    for candidate in "$@"; do
+        [ -f "$candidate" ] && realpath "$candidate"
+    done | awk '!x[$0]++' | # remove duplicates
+    while read -r tags; do
+        tagroot=$(realpath "${tags%/*}")
+        readtags -t "$tags" "$tagname" |
+        while read tag; do
+            name=$(printf "%s\n" "$tag" | cut -f 2 | sed "s:':'':g")
+            menuinfo=$(printf "%s\n" "$tag" | sed "s:.*/\^\(\s\+\)\?::;s:\(\\\$\)\?/$::;s:':'':g;s:$open:\\\\$open:g")
+            keys=$(printf "%s\n" "$tag" | sed "s:.*/\^::;s:\(\\\$\)\?/$::;s:':'''''''''''''''':g;s:<:<lt>:g;s:\\t:<c-v><c-i>:g")
+            file=$(printf "%s\n" "$name" | sed "s:'':'''''''''''''''':g")
+            tagroot=$(printf "%s\n" "$tagroot" | sed "s:':'''''''''''''''':g")
+            select=$(printf "%s\n" "$tagname" | sed "s:':'''''''''''''''':g;s:<:<lt>:g;s:\\t:<c-v><c-i>:g")
+            command="evaluate-commands '' try '''' edit ''''''''$tagroot/$file''''''''; execute-keys ''''''''/\Q$keys<ret>vcs\Q$select<ret>'''''''' '''' catch '''' echo -markup ''''''''{Error}unable to find tag'''''''' '''' ''"
+            if [ -n "$file" ] && [ -n "$keys" ]; then
+                printf "%s " "'$name {MenuInfo}$menuinfo' '$command'" >> $menu
             fi
-            cat "$namecache"
-        done} \
-    -docstring %{ctags-search [<symbol>]: jump to a symbol's definition
-If no symbol is passed then the current selection is used as symbol name} \
-    ctags-search \
-    %{ evaluate-commands %sh{
-        realpath() { ( path=$(readlink "$1"); cd "$(dirname "$1")"; printf "%s/%s\n" "$(pwd -P)" "$(basename "$1")" ) }
-        export tagname=${1:-${kak_selection}}
-        eval "set -- $kak_opt_ctagsfiles"
-        for candidate in "$@"; do
-            [ -f "$candidate" ] && realpath "$candidate"
-        done | awk '!x[$0]++' | # remove duplicates
-        while read -r tags; do
-            printf '!TAGROOT\t%s\n' "$(realpath "${tags%/*}")/"
-            readtags -t "$tags" $tagname
-        done | awk -F '\t|\n' '
-        /^!TAGROOT\t/ { tagroot=$2 }
-        /[^\t]+\t[^\t]+\t\/\^.*\$?\// {
-            re=$0;
-            sub(".*\t/\\^", "", re); sub("\\$?/$", "", re); gsub("(\\{|\\}|\\\\E).*$", "", re);
-            keys=re; gsub(/</, "<lt>", keys); gsub(/\t/, "<c-v><c-i>", keys);
-            out = out " %{" $2 " {MenuInfo}" re "} %{evaluate-commands %{ try %{ edit %{" tagroot $2 "}; execute-keys %{/\\Q" keys "<ret>vc} } catch %{ echo %{unable to find tag} } } }"
-        }
-        /[^\t]+\t[^\t]+\t[0-9]+/ { out = out " %{" $2 ":" $3 "} %{evaluate-commands %{ edit %{" tagroot $2 "} %{" $3 "}}}" }
```

Comment:
Tags with line numbers are not practical, because in some languages, often several tags are located on the same line. In my opinion tag jump should bring us directly to the tag itself, as it is done in Vim and Emacs, and in every other editor that has `ctags` or `gnu global` support. This is more reliable. Although I don't know if it is possible to have absolutely identical tags in the same file but on a different lines, where filtering by line will be actually useful, but I think that this will be covered by the first search pattern that searches from the beginning of the line.